tests/cupy_tests/indexing_tests/test_generate.py

```diff
@@ -40,7 +39,6 @@ class TestIX_(unittest.TestCase):
     def test_ix_list(self, xp):
         return xp.ix_([0, 1], [2, 4])
 
-    @pytest.mark.xfail(runtime.is_hip, reason='HIP may have a bug')
```

Comment:
We might need to keep this around for ROCm 4.0...
```
FAILED cupy_tests/indexing_tests/test_generate.py::TestIX_::test_ix_ndarray
```
https://jenkins.preferred.jp/job/chainer/job/cupy_pr/3978/TEST=rocm-4-0,label=mnj-mi50/consoleText

Reply:
> We might need to keep this around for ROCm 4.0...

I noticed something weird. In the two failing tests
```
FAILED cupy_tests/creation_tests/test_from_data.py::TestFromData::test_array_from_list_of_numpy_scalar
FAILED cupy_tests/indexing_tests/test_generate.py::TestIX_::test_ix_ndarray
```
The returned CuPy array (`cupy_result`) actually agreed with NumPy's (`numpy_result`). @toslunar Any chance you know off top of your head what could be wrong? Is it because the ROCm 4.0 uses an older NumPy (1.19)?

Reply:
At `TestFromData.test_array_from_list_of_numpy_scalar`, PyTest shows local variables are
```
x = array([0, 1], dtype=int16), y = array([0, 1], dtype=int16), err_msg = ''
verbose = True, strides_check = True
```
while the message in the exception contains
```
E        x: array([0, 0], dtype=int16)
E        y: array([0, 1], dtype=int16)
```
So, there seems a gpu sync error. (CuPy array is updated between the assertion and PyTest's inspect.)

Reply:
This failure is reproducible on ROCm 4.0 but not seen on 4.2:
https://jenkins.preferred.jp/job/chainer/job/cupy_pr/4026/TEST=rocm-4-0,label=mnj-mi50/consoleText
Maybe something was fixed during 4.0 -> 4.2...